siac/parse.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+)
+
+// parseSize converts strings of form 10GB to a size in bytes
+// Parsing is flexible in that unnecessary trailing characters are discarded
+// If parsing fails an error is thrown
+func parseSize(strSize string) (string, error) {
+	// if the value is at least 2 or 3 characters it may have a suffix
+	// to detect a suffix a check is made that the character is outside
+	// the ascii range for numbers (48-57), letters are (65-90, 97-122)
```

Comment:
comments, in particular long comments (exceptions can be made for shorter comments) should have full punctuation, including capitals and periods.